File: ios/Classes/PushyPlugin.m
#import "PushyPlugin.h"
#import <pushy_flutter/pushy_flutter-Swift.h>

@implementation PushyPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [PushyFlutter registerWithRegistrar:registrar];
}
@end

File: ios/pushy_flutter/Sources/pushy_flutter/PushyPlugin.swift
import Flutter
import Foundation

// Expose the class to Objective-C under the name `PushyPlugin` so Flutter's
// generated plugin registrant can discover it via the iOS `pluginClass` entry.
@objc(PushyPlugin)
public class PushyPlugin: NSObject, FlutterPlugin {
    // Plugin entry point invoked by Flutter's generated registrant
    public static func register(with registrar: FlutterPluginRegistrar) {
        // Delegate registration to the actual implementation
        PushyFlutter.register(with: registrar)
    }
}
